<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1045</width>
    <height>631</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>qudi: HFML spectroscopy</string>
  </property>
  <widget class="QWidget" name="centralwidget"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1045</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="viewAction"/>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <addaction name="menuView"/>
  </widget>
  <widget class="QDockWidget" name="spectrumDockWidget">
   <property name="windowTitle">
    <string>Spectrum</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout">
     <item row="2" column="1" colspan="2">
      <widget class="PlotWidget" name="spectrum_PlotWidget"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="controlToolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>Controls</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="focus_Action"/>
   <addaction name="acquisition_Action"/>
   <addaction name="save_Action"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="windowTitle">
    <string>CCD parameters</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <widget class="QDoubleSpinBox" name="focus_doubleSpinBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>71</width>
       <height>22</height>
      </rect>
     </property>
     <property name="keyboardTracking">
      <bool>false</bool>
     </property>
     <property name="value">
      <double>1.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="acquisition_doubleSpinBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>71</width>
       <height>22</height>
      </rect>
     </property>
     <property name="keyboardTracking">
      <bool>false</bool>
     </property>
     <property name="maximum">
      <double>36000.000000000000000</double>
     </property>
     <property name="value">
      <double>1.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="focus_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Focus (s):</string>
     </property>
    </widget>
    <widget class="QLabel" name="acquisition_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Aquisition (s):</string>
     </property>
    </widget>
    <widget class="QLabel" name="number_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Number of spectra:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="number_of_spectra_spinBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>71</width>
       <height>22</height>
      </rect>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <widget class="QLabel" name="ROI_label">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>0</y>
       <width>101</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>ROI</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSpinBox" name="roi_x0_spinBox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>40</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>9999</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="roi_x_max_spinBox">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>40</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>9999</number>
     </property>
     <property name="value">
      <number>1340</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="roi_y0_spinBox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>60</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>9999</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="roi_y_max_spinBox">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>60</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>9999</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
    </widget>
    <widget class="QCheckBox" name="bin_checkBox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>100</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Hardware binning of all the pixels along y axis.&lt;/p&gt;&lt;p&gt;Displays data as one-dimensional spectrum.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>Bin y?</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="flip_x_checkBox">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>100</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Flips (mirrors) data from the ccd in x direction (horizontal).</string>
     </property>
     <property name="text">
      <string>Flip x?</string>
     </property>
    </widget>
    <widget class="QLabel" name="ROI_label_x">
     <property name="geometry">
      <rect>
       <x>141</x>
       <y>40</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>x</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="ROI_label_y">
     <property name="geometry">
      <rect>
       <x>141</x>
       <y>60</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>y</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="ROI_label_min">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>20</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>min</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="ROI_label_max">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>20</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>max</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QComboBox" name="energy_selector_comboBox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>130</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Pixel</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Wavelength (nm)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Raman shift (cm-1)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Energy (eV)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Energy (meV)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Wavenumber (cm-1)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Frequency (THz)</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="constant_background_label">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>160</y>
       <width>161</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Constant background (counts):</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="constant_background_spinBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>180</y>
       <width>61</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>5000</number>
     </property>
    </widget>
    <widget class="QLabel" name="adc_quality_label">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>11</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>AdcQuality:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="adc_quality_comboBox">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>10</y>
       <width>61</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="adc_analog_gain_label">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>41</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>AdcAnalogGain:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="adc_analog_gain_comboBox">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>40</y>
       <width>61</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="shutter_timing_mode_label">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>71</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>ShutterTimingMode:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="shutter_timing_mode_comboBox">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>70</y>
       <width>61</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="adc_speed_label">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>101</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>AdcSpeed (MHz):</string>
     </property>
    </widget>
    <widget class="QComboBox" name="adc_speed_comboBox">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>100</y>
       <width>61</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="vertical_shift_rate_label">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>131</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>VerticalShiftRate (us):</string>
     </property>
    </widget>
    <widget class="QComboBox" name="vertical_shift_rate_comboBox">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>130</y>
       <width>61</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="image2DockWidget">
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <widget class="QWidget" name="widget" native="true">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>30</y>
       <width>921</width>
       <height>211</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <action name="focus_Action">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>../../artwork/icons/qudiTheme/22x22/start-counter.png</normaloff>
     <normalon>../../artwork/icons/qudiTheme/22x22/stop-counter.png</normalon>../../artwork/icons/qudiTheme/22x22/start-counter.png</iconset>
   </property>
   <property name="text">
    <string>Focus</string>
   </property>
   <property name="toolTip">
    <string>Focus</string>
   </property>
  </action>
  <action name="save_Action">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>../../artwork/icons/qudiTheme/22x22/record-counter.png</normaloff>../../artwork/icons/qudiTheme/22x22/record-counter.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save count trace to file</string>
   </property>
  </action>
  <action name="viewAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Display</string>
   </property>
   <property name="toolTip">
    <string>Show the Data display</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset>
     <normaloff>../../artwork/icons/oxygen/22x22/application-exit.png</normaloff>../../artwork/icons/oxygen/22x22/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Close</string>
   </property>
  </action>
  <action name="acquisition_Action">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>../../artwork/icons/qudiTheme/22x22/goto-poi.png</normaloff>../../artwork/icons/qudiTheme/22x22/goto-poi.png</iconset>
   </property>
   <property name="text">
    <string>Acquisition</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>viewAction</sender>
   <signal>triggered(bool)</signal>
   <receiver>spectrumDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>136</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spectrumDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>viewAction</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
